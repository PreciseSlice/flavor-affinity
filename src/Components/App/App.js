import React, { Component } from 'react';
import { connect } from 'react-redux';
import './App.css';
import MainForm from '../MainForm/MainForm';
import CardContainer from '../CardContainer/CardContainer';
//import { getAllIngredients } from '../../Helpers/apiCalls';
import { setAllIngredients } from '../../Actions';
//import * as actions from '../../Actions';
import { allDataClean } from '../../Helpers/.mockApi.js';
//import { cleanData } from '../App/testData';

export class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      error: null,
      errorInfo: null
    };
  }

  async componentDidMount() {
    const { setAllIngredients } = this.props;
    //const allIngredients = await getAllIngredients();
    //setAllIngredients(allIngredients);
    setAllIngredients(allDataClean);
  }

  componentDidCatch(error, errorInfo) {
    this.setState({
      error: error,
      errorInfo: errorInfo
    });
  }

  clearError() {
    this.setState({
      error: null,
      errorInfo: null
    });
  }

  render() {
    if (this.state.error) {
      return (
        <div className="eror">
          <h2>{'Oh-no! Something went wrong'}</h2>
          <p className="">{this.state.error && this.state.error.toString()}</p>
          <div>{'Component Stack Error Details: '}</div>
          <p className="">{this.state.errorInfo.componentStack}</p>
          <button onClick={() => this.clearError()} />
        </div>
      );
    }
    return (
      <div className="app">
        <MainForm />
        <CardContainer />
      </div>
    );
  }
}

// export const mapStateToProps = state => ({

// });

export const mapDispatchToProps = dispatch => ({
  setAllIngredients: ingredient => dispatch(setAllIngredients(ingredient))
});

export default connect(null, mapDispatchToProps)(App);
