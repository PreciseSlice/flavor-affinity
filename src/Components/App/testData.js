/* eslint-disable */
export const allIngredients = [
  {
    id: 5902,
    name: '1724 Tonic',
    product: '1724 Tonic',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 4578,
        src: 'https://pantry.foodpairing.com/apps/4578',
        size_240: 'https://pantry.foodpairing.com/apps/4578_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/4578_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/4578_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/4578_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/4578_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/4578_1440_0'
      },
      brand: null,
      categories: [
        {
          id: 38,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/38'
        },
        {
          id: 139,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/139'
        }
      ]
    },
    description:
      'The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 0
    }
  },
  {
    id: 5440,
    name: 'A la Chimay Bleue',
    product: 'A la Chimay Bleue',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 1772,
        src: 'https://pantry.foodpairing.com/apps/1772',
        size_240: 'https://pantry.foodpairing.com/apps/1772_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/1772_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/1772_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/1772_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/1772_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/1772_1440_0'
      },
      brand: {
        id: 80,
        name: 'Chimay',
        website: '',
        description: '',
        _links: {
          images: [
            {
              id: 4420,
              src: 'https://pantry.foodpairing.com/apps/4420'
            }
          ]
        }
      },
      categories: [
        {
          id: 265,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/265'
        },
        {
          id: 266,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/266'
        }
      ]
    },
    description:
      'The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 1
    }
  },
  {
    id: 5443,
    name: 'A la Chimay Rouge',
    product: 'A la Chimay Rouge',
    preparation: '',
    _links: {
      preparation: null,
      origin: null,
      image: {
        id: 1773,
        src: 'https://pantry.foodpairing.com/apps/1773',
        size_240: 'https://pantry.foodpairing.com/apps/1773_240_0',
        size_480: 'https://pantry.foodpairing.com/apps/1773_480_0',
        size_720: 'https://pantry.foodpairing.com/apps/1773_720_0',
        size_960: 'https://pantry.foodpairing.com/apps/1773_960_0',
        size_1200: 'https://pantry.foodpairing.com/apps/1773_1200_0',
        size_1440: 'https://pantry.foodpairing.com/apps/1773_1440_0'
      },
      brand: {
        id: 80,
        name: 'Chimay',
        website: '',
        description: '',
        _links: {
          images: [
            {
              id: 4420,
              src: 'https://pantry.foodpairing.com/apps/4420'
            }
          ]
        }
      },
      categories: [
        {
          id: 265,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/265'
        },
        {
          id: 266,
          href:
            'https://api.foodpairing.com/taxonomies/ingredients/categories/266'
        }
      ]
    },
    description:
      'The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.',
    _meta: {
      created_at: '',
      modified_at: '',
      index: 2
    }
  }
];

export const cleanData = [
    {
      id: 5902,
      name: '1724 tonic',
      image: 'https://pantry.foodpairing.com/apps/4578_240_0',
      description:
        'The essence of 1724 Tonic Water has its origins in the Andes, on the mythical Inca Trail. This is not by chance. This is where quinine was discovered. There, this ingredient gains the authenticity of a product with roots in the region that go back hundreds of years. At 1724 metres above sea level. Not a metre above or below.'
    },
    {
      id: 5440,
      name: 'a la chimay bleue',
      image: 'https://pantry.foodpairing.com/apps/1772_240_0',
      description:
        'The rind of the à la Chimay Bleue is regularly washed with the Chimay Blue Cap during his stay in the aging cellars of the monestarey. Pronounced flavor with a slight bitterness of the hops.'
    },
    {
      id: 5443,
      name: 'a la chimay rouge',
      image: 'https://pantry.foodpairing.com/apps/1773_240_0',
      description:
        'The à la Chimay Rouge is a round, smooth cheese with creamy texture. Its natural rind is washed with Chimay Red Cap. Fruity flavour with slight bitterness.'
    }
];
