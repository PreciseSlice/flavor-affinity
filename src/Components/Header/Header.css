.header-container {
  align-items: center;
  background: rgba(87, 87, 87, 0.6);
  box-shadow: 0 0 5px 5px rgba(87, 87, 87, 0.6);
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  color: white; }
  .header-container h1 {
    font-size: 7em;
    letter-spacing: 0.2em; }
  .header-container h2 {
    font-size: 2.5em;
    margin: 0;
    font-weight: 400; }
  .header-container h3 {
    font-size: 2em;
    margin: 10px 0 0 0;
    font-weight: 400; }
  .header-container p {
    font-size: 1.6em; }

.header-btn-container {
  align-items: center;
  display: flex;
  justify-content: space-evenly;
  margin: -5px 0 30px 0;
  width: 58%; }
  .header-btn-container a {
    text-decoration: none; }
  .header-btn-container button {
    width: 22%; }
  .header-btn-container button:hover {
    background: rgba(81, 102, 85, 0.9);
    border: none;
    color: white; }
    .header-btn-container button:hover a {
      background: rgba(81, 102, 85, 0.9);
      color: white; }
