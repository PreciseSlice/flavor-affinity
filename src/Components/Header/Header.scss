.header-container {
  align-items: center;
  background: rgba(87, 87, 87, 0.6);
  box-shadow: 0 0 5px 5px  rgba(87, 87, 87, 0.6);
  display: flex;
  justify-content: center;
  flex-direction: column;
  width: 100%;
  color: white;

  h1 {
    font-size: 8em;
    letter-spacing: 0.2em;
  }

  p{
    font-size: 1.6em;
  }
}

.header-btn-container {
  align-items: center;
  display: flex;
  justify-content: space-evenly;
  margin-top: -15px;
  width: 56%;

  a {
   text-decoration: none; 
  }

  button {
    width: 21%;
  }

  button:hover {
    background: rgba(81, 102, 85, 0.9);
    border: none;
    color: white;
    a {
      background: rgba(81, 102, 85, 0.9);
      color: white;
    }
  }

}