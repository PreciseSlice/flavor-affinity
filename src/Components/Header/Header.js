import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import './Header.css';
import { clearSelectedCards, clearPairings } from '../../Actions/index';
import ReactTooltip from 'react-tooltip';

export const Header = ({
  selectedCards,
  clearSelectedCards,
  clearPairings
}) => {
  return (
    <div className="header-container">
      <ReactTooltip />
      <h1>blend</h1>
      <h2>serving inspiration daily</h2>
      <h3>discover ingredient pairings instantly </h3>
      <div className="header-btn-container">
        <button
          onClick={() => {
            clearSelectedCards();
            clearPairings();
          }}
        >
          <NavLink to="/">home</NavLink>
        </button>
        <button data-tip="compare selected ingredients">
          <NavLink to="/selected">ingredients: {selectedCards.length}</NavLink>
        </button>
        <button
          onClick={() => {
            clearPairings();
          }}
        >
          clear pairings
        </button>
        <button
          onClick={() => {
            clearSelectedCards();
          }}
        >
          clear selections
        </button>
      </div>
    </div>
  );
};

export const mapStateToProps = state => ({
  selectedCards: state.selectedCards
});

export const mapDispatchToProps = dispatch => ({
  clearSelectedCards: () => dispatch(clearSelectedCards()),
  clearPairings: () => dispatch(clearPairings())
});

export default connect(mapStateToProps, mapDispatchToProps)(Header);

Header.propTypes = {
  selectedCards: PropTypes.oneOfType([
    PropTypes.array,
    PropTypes.arrayOf(
      PropTypes.shape({
        description: PropTypes.string,
        id: PropTypes.number.isRequired,
        image: PropTypes.string.isRequired,
        name: PropTypes.string.isRequired,
        selected: PropTypes.bool.isRequired
      })
    )
  ]),

  clearSelectedCards: PropTypes.func.isRequired,
  clearPairings: PropTypes.func.isRequired
};
