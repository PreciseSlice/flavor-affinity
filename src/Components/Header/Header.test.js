import React from 'react';
import ReactDOM from 'react-dom';
import { Header , mapStateToProps } from './Header';
import { shallow } from 'enzyme';
import { cleanData } from '../App/testData';

describe('Header', () => {
  let wrapper;
  const mockFn = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Header selectedCards={cleanData} />);
  });

  it('matches snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('mapStateToProps', () => {
    it('should map items in the store to props', () => {
      const mockStore = {
        state: {
          selectedCards: cleanData
        }
      };
      const mapped = mapStateToProps(mockStore);

      expect(mapped.selectedCards).toEqual(mockStore.selectedCards);
    });
  });

});