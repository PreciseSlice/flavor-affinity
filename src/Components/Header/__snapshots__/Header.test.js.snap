// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header matches snapshot 1`] = `
<div
  className="header-container"
>
  <h1>
    blend
  </h1>
  <div
    className="header-btn-container"
  >
    <button>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/"
      >
        Home
      </NavLink>
    </button>
    <button>
      <NavLink
        activeClassName="active"
        ariaCurrent="true"
        to="/selected"
      >
        Ingredients: 
        3
      </NavLink>
    </button>
  </div>
</div>
`;
